src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java

```diff
@@ -293,23 +301,65 @@ private Target loadBuildSetting(String targetToBuild) throws OptionsParsingExcep
       return buildSettings.get(targetToBuild);
     }
 
-    Target buildSetting;
-    try {
-      buildSetting = buildSettingLoader.loadBuildSetting(targetToBuild);
-      if (buildSetting == null) {
-        return null;
+    Target target;
+    String targetToLoadNext = targetToBuild;
+    SequencedSet<Label> aliasChain = new LinkedHashSet<>();
+    while (true) {
+      try {
+        target = buildSettingLoader.loadBuildSetting(targetToLoadNext);
+        if (target == null) {
+          return null;
+        }
+      } catch (InterruptedException | TargetParsingException e) {
+        Thread.currentThread().interrupt();
+        throw new OptionsParsingException(
+            "Error loading option " + targetToBuild + ": " + e.getMessage(), targetToBuild, e);
+      }
+      if (!aliasChain.add(target.getLabel())) {
+        throw new OptionsParsingException(
+                String.format(
+                        "Failed to load build setting '%s' due to a cycle in alias chain: %s",
+                        targetToBuild,
+                        formatAliasChain(Stream.concat(aliasChain.stream(), Stream.of(target.getLabel())))),
+                targetToBuild);
+      }
+      if (target.getAssociatedRule() == null) {
+        throw new OptionsParsingException(
+                String.format("Unrecognized option: %s", formatAliasChain(aliasChain.stream())),
+                targetToBuild);
+      }
+      if (target.getAssociatedRule().isBuildSetting()) {
+        break;
+      }
+      // Follow the unconfigured values of aliases.
+      if (target.getAssociatedRule().getRuleClass().equals(ALIAS_RULE_NAME)) {
+        targetToLoadNext = switch (target.getAssociatedRule().getAttr(ALIAS_ACTUAL_ATTRIBUTE_NAME)) {
+          case Label label -> label.getUnambiguousCanonicalForm();
+          case BuildType.SelectorList<?> ignored -> throw new OptionsParsingException(
+                  String.format(
+                          "Failed to load build setting '%s' as it resolves to an alias with an actual"
+                                  + " value that uses select(), which is not supported: %s",
```

Comment:
It'd be nice if there was some concise wording that resonates with users about why this isn't supported (vs. a mysterious "this just doesn't work here"). Something about how we're still figuring out the configuration at this stage so it's too early to reason around it? Any way to integrate a bit more rationale?

Reply:
Good point, added an attempt at an explanation.